_posts/2024-XX-XX-quarkus-kiota.adoc

```diff
@@ -0,0 +1,165 @@
+This is the motivation behind the birth of the new ``quarkus-kiota`` extension.
+The project is in its early stages, and we encourage you to try it out and provide feedback.
+
+The codebase lives in the link:https://github.com/quarkiverse/quarkus-kiota[Quarkiverse], the project is listed in the link:https://quarkus.io/extensions/io.quarkiverse.kiota/quarkus-kiota/[extensions] and the link:https://docs.quarkiverse.io/quarkus-kiota/dev/index.html[docs] are available at the usual location.
+
+Get started by adding this extension:
+
+----
+quarkus ext add io.quarkiverse.kiota:quarkus-kiota
+----
+
+Since it's a code generator extension, you will need to add the `generate-code` goal to the `quarkus-maven-plugin` `executions` section:
+
+[source,xml]
+----
+<plugin>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-maven-plugin</artifactId>
+  <extensions>true</extensions>
+  <executions>
+    <execution>
+      <goals>
+        <goal>build</goal>
+        <goal>generate-code</goal>
+      </goals>
+    </execution>
+  </executions>
+</plugin>
+----
+
+Add the necessary dependencies, we are going to replace the default http and json serialization libraries published by Microsoft and swap in Vert.X and Jackson(published from this https://github.com/kiota-community/kiota-java-extra[repository]) as they nicely play with the rest of the stack of a typical Quarkus based application.
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.microsoft.kiota</groupId>
+  <artifactId>microsoft-kiota-abstractions</artifactId>
+  <version>${kiota.libs.version}</version>
+</dependency>
+<dependency>
+  <groupId>io.kiota</groupId>
+  <artifactId>kiota-http-vertx</artifactId> <!-- alternatively <artifactId>kiota-http-jdk</artifactId> -->
+  <version>{kiota-java-extra.version}</version>
+</dependency>
+<dependency>
+  <groupId>io.kiota</groupId>
+  <artifactId>kiota-serialization-jackson-quarkus</artifactId>
+  <version>{kiota-java-extra.version}</version>
+</dependency>
+<dependency>
+  <groupId>com.microsoft.kiota</groupId>
+  <artifactId>microsoft-kiota-serialization-text</artifactId>
+  <version>${kiota.libs.version}</version>
+</dependency>
+<dependency>
+  <groupId>com.microsoft.kiota</groupId>
+  <artifactId>microsoft-kiota-serialization-form</artifactId>
+  <version>${kiota.libs.version}</version>
+</dependency>
+<dependency>
+  <groupId>com.microsoft.kiota</groupId>
+  <artifactId>microsoft-kiota-serialization-multipart</artifactId>
+  <version>${kiota.libs.version}</version>
+</dependency>
+<dependency>
+  <groupId>jakarta.annotation</groupId>
+  <artifactId>jakarta.annotation-api</artifactId>
+</dependency>
+----
+
+Now we need to generate the actual client for our OpenAPI description, to do so, you should drop the OpenAPI file(in ``yaml`` or ``json`` format) in the ``src/main/openapi`` folder of your project.
+You are all set to use the client in your application!
+
+[source,java]
+----
```

Comment:
would be good to have the imports visible here - just for full transparency.

Reply:
👍

Reply:
Done here:
https://github.com/quarkusio/quarkusio.github.io/pull/1885/commits/850a8a6a8538f985309786070feba4b0668e2866

More specifically:
`io.apisdk.example.yaml.ApiClient`: is the generated code
`io.kiota.http.vertx.VertXRequestAdapter`: is the `RequestAdapter` implementation we are shipping [here](https://github.com/kiota-community/kiota-java-extra/tree/main?tab=readme-ov-file#http-vertx)